timesketch/lib/analyzers/feature_plugins/winevt.py

```diff
@@ -0,0 +1,172 @@
+            # NOTE: feature_name and feature_config value checking is done at the sink
+            # function `extract_features`.
+            feature_name = feature.get("feature", None)
+            feature_config = feature.get("feature_config", None)
+
+            result = self.extract_features(feature_name, feature_config)
+            if not result:
+                logger.debug("No result for plugin run for %s", feature_name)
+            results.append(result)
+
+        return "\n".join(results)
+
+    def extract_features(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Features extraction name.
+            config (dict): A dict that contains the configuration fo the
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+
+        if not name:
+            raise ValueError("Feature name is empty")
+
+        if not config:
+            raise ValueError("Feature configuration value is empty")
+
+        source_name = config.get("source_name", None)
+        if not source_name:
+            raise ValueError("source_name is empyt")
+
+        if not isinstance(source_name, list):
+            raise TypeError(
+                f"Expecting source_name as a list in {name} and got {type(source_name)}"
+            )
+
+        source_name = source_name[0]
+
+        # Using -1 as a default value for event_identifier and event_version as
+        # 0 is considered as None.
+        event_identifier = int(config.get("event_identifier", -1))
+        event_version = int(config.get("event_version", -1))
+
+        if not source_name:
+            raise ValueError("source_name is empty")
+
+        if event_identifier < 0:
+            raise ValueError(
+                f"event_identifier is {event_identifier}. Expecting the value >= 0"
+            )
+
+        if event_version < 0:
+            raise ValueError(f"event_version is {event_version}. Expecting value >= 0")
+
+        mappings = config.get("mapping", None)
+        if not mappings:
+            raise ValueError(f"mapping value for {name} is empty")
+
+        query = (
+            f"source_name: {source_name} AND event_identifier: {event_identifier}"
+            f" AND event_version: {event_version}"
+        )
+
+        events = self.analyzer_object.event_stream(
+            query_string=query, return_fields=self.EVENT_FIELDS
+        )
+        event_counter = 0
+
+        for event in events:
+            attributes = {}
+
+            strings = event.source.get("strings", None)
+            if not strings:
+                logger.debug("No strings attribute in event ID: %s", event.event_id)
+                continue
+
+            for mapping in mappings:
+                attribute_name = mapping.get("name", None)
+                string_index = int(mapping.get("string_index", -1))
```

Comment:
1. The mapping fields `name` and `string_index` are mandatory. If there is no definition this mapping should be skipped. 
2. Using `-1` here as the default value for `string_index` will result in an unexpected behavior in line 127. Since `strings[-1]` will return the last element of the string.

Reply:
Done. Checking for `name` and `string_index`  in `winevt_features.py#L139` and ``winevt_features.py#L145`.